get-resource.sh

```diff
@@ -39,41 +37,41 @@ TMPDIR="$(mktemp -d -p "${SHARED_DIR}"/tmp)"
 # If we have a CACHEURL and nothing has yet been downloaded
 # get header info from the cache
 
-if [ -n "${CACHEURL:-}" ] && [ ! -e "${FFILENAME}.headers" ]; then
-    curl_with_flags -g --fail -O "${CACHEURL}/${FFILENAME}.headers" || true
+if [ -n "${CACHEURL:-}" ] && [ ! -e "${FILENAME}.headers" ]; then
+    curl_with_flags -g --fail -O "${CACHEURL}/${FILENAME}.headers" || true
 fi
 
 # Download the most recent version of IPA
 if [ -r "${DESTNAME}.headers" ] ; then
     ETAG="$(awk '/ETag:/ {print $2}' "${DESTNAME}.headers" | tr -d "\r")"
     cd "${TMPDIR}"
-    curl_with_flags -g --dump-header "${FFILENAME}.headers" \
-        -O "${IPA_BASEURI}/${FFILENAME}" \
-        --header "If-None-Match: ${ETAG}" || cp "${SHARED_DIR}/html/images/${FFILENAME}.headers" .
+    curl_with_flags -g --dump-header "${FILENAME}.headers" \
+        -O "${IPA_BASEURI}/${FILENAME}" \
+        --header "If-None-Match: ${ETAG}" || cp "${SHARED_DIR}/html/images/${FILENAME}.headers" .
 
     # curl didn't download anything because we have the ETag already
     # but we don't have it in the images directory
     # Its in the cache, go get it
-    ETAG="$(awk '/ETag:/ {print $2}' "${FFILENAME}.headers" | tr -d "\"\r")"
-    if [ ! -s "${FFILENAME}" ] && [ ! -e "${SHARED_DIR}/html/images/${FILENAME}-${ETAG}/${FFILENAME}" ]; then
-        mv "${SHARED_DIR}/html/images/${FFILENAME}.headers" .
-        curl_with_flags -g -O "${CACHEURL}/${FILENAME}-${ETAG}/${FFILENAME}"
+    ETAG="$(awk '/ETag:/ {print $2}' "${FILENAME}.headers" | tr -d "\"\r")"
+    if [ ! -s "${FILENAME}" ] && [ ! -e "${SHARED_DIR}/html/images/${FILENAME%%.*}-${ETAG}/${FILENAME}" ]; then
+        mv "${SHARED_DIR}/html/images/${FILENAME}.headers" .
+        curl_with_flags -g -O "${CACHEURL}/${FILENAME}-${ETAG}/${FILENAME}"
```

Comment:
The first instance of FILENAME needs the same treatment here (maybe add a FILENAME_NO_EXT variable for convenience?)